src/scripts/nlc.confidence.high.js

```diff
@@ -47,50 +47,53 @@ module.exports = function(robot) {
-			robot.logger.info(`${TAG} High confidence detected.`);
 			handle(db, res, classification, robot);
 		}).catch((err) => {
-			robot.logger.error(err);
+			robot.emit('ibmcloud.formatter', { response: res, message: i18n.__('nlc.error.unexpected.general')});
+			robot.logger.error(`${TAG}: Error processing high confidence NLC for ${classification.top_class}. Error=${err}`);
 		});
 	});
 
 
 	function handle(db, res, classification, robot){
-		// TODO: Remove displaying of top classes and confidence back to user.
-		res.send(i18n.__('nlc.confidence.high.process', classification.top_class));
-
-		// Record classification success for metrics.
-		db.post(classification, 'classified', classification.top_class).then(() => {
-
-			// Call emit target if specified with parameter values
-			nlcconfig.getClassEmitTarget(classification.top_class).then((tgt) => {
-				if (tgt) {
-					// Obtain statement from res removing the bot name
-					var text = res.message.text.replace(robot.name, '').trim();
-					paramManager.getParameters(classification.top_class, text, tgt.parameters).then((parameters) => {
-						extractParameters.validateParameters(robot, res, paramManager, classification.top_class, text, tgt.parameters, parameters).then((validParameters) => {
-							var authEmitParams = {
-								emitTarget: tgt.target,
-								emitParameters: validParameters
-							};
-							robot.logger.info(`${TAG} Emitting to NLC target ${tgt.target} with params=${validParameters}`);
-							robot.emit('ibmcloud-auth-to-nlc', res, authEmitParams);
-						}).catch(function(error) {
-							robot.logger.error(`${TAG} Error occurred trying to obtain parameters for top class; top class = ${classification.top_class}; text = ${text}; error = ${error}.`);
-						});
+		// Call emit target if specified with parameter values
+		nlcconfig.getClassEmitTarget(classification.top_class).then((tgt) => {
+			if (tgt) {
+				// Obtain statement from res removing the bot name
+				var text = res.message.text.replace(robot.name, '').trim();
+				paramManager.getParameters(classification.top_class, text, tgt.parameters).then((parameters) => {
+					extractParameters.validateParameters(robot, res, paramManager, classification.top_class, text, tgt.parameters, parameters).then((validParameters) => {
+						var authEmitParams = {
+							emitTarget: tgt.target,
+							emitParameters: validParameters
+						};
+						robot.logger.info(`${TAG} Emitting to NLC target ${tgt.target} with params=${validParameters}`);
+						robot.emit('ibmcloud-auth-to-nlc', res, authEmitParams);
 					}).catch(function(error) {
-						robot.logger.error(`${TAG} Error occurred trying to obtain parameters for top class; top class = ${classification.top_class}; text = ${text}; error = ${error}.`);
+						robot.logger.error(`${TAG} Error occurred trying to validate parameters for top class; top class = ${classification.top_class}; text = ${text}; error = ${error}.`);
+						throw error; // To report to user
```

Comment:
The behavior you described is what I was looking for, but I see it could be confusing and maybe not the best choice, so instead of throwing the error, I'm reporting it to the user when it happens.